pinot-common/src/main/java/org/apache/pinot/common/function/DateTimePatternHandler.java

```diff
@@ -35,15 +35,23 @@ private DateTimePatternHandler() {
    */
   public static long parseDateTimeStringToEpochMillis(String dateTimeString, String pattern) {
     DateTimeFormatter dateTimeFormatter = getDateTimeFormatter(pattern);
-    return dateTimeFormatter.parseMillis(dateTimeString);
+    return parseMillis(dateTimeString, dateTimeFormatter);
   }
 
   /**
    * Converts the dateTimeString of passed pattern into a long of the millis since epoch
    */
   public static long parseDateTimeStringToEpochMillis(String dateTimeString, String pattern, String timezoneId) {
     DateTimeFormatter dateTimeFormatter = getDateTimeFormatter(pattern, timezoneId);
-    return dateTimeFormatter.parseMillis(dateTimeString);
+    return parseMillis(dateTimeString, dateTimeFormatter);
+  }
+
+  private static long parseMillis(String dateTimeString, DateTimeFormatter dateTimeFormatter) {
+    try {
+      return dateTimeFormatter.parseMillis(dateTimeString);
+    } catch (Exception e) {
+      return 0L;
```

Comment:
i have concerns on doing this by default. one alternative is to add a 
```
fromDateTime(String dateTimeString, String pattern, long defaultValue)
```
and only allow returning default value when malformed dateTimeString is encountered otherwise still throw an exception. please share your thoughts

Reply:
I'd suggest doing it more explicitly by asking for an extra default value. This is similar to how we handle json path error

Reply:
yeah unfortunately the function signature with extra default value is taken by the timezone :-( so we can't add that

Reply:
Why? The default value should be `long` type

Reply:
function registry is using number of arguments to match scalar function, it doesn't consider type b/c v1 is not-strong typed